include/seastar/core/sharded.hh

```diff
@@ -52,6 +52,8 @@
 
 namespace seastar {
 
+extern logger seastar_logger;
```

Comment:
why? i can still compile seastar without this declaration.

Reply:
I agree - sharded.hh doesn't need seastar_logger for itself. If existing code needed seastar_logger and got it accidentally (because reactor.hh), I don't think we need to support this - because the same can be true for basically anything which is defined in reactor.hh, not just seastar_logger.

By the way, I a natural place to put that "extern logger" thing is  <seastar/util/log.hh>. Since log.hh is still included in sharded.hh, it will also solve the problem for the user of sharded.hh who assumed that seastar_logger is defined.

Reply:
> I agree - sharded.hh doesn't need seastar_logger for itself. If existing code needed seastar_logger and got it accidentally (because reactor.hh), I don't think we need to support this - because the same can be true for basically anything which is defined in reactor.hh, not just seastar_logger.

I had a failure, will check again.

> 
> By the way, I a natural place to put that "extern logger" thing is <seastar/util/log.hh>. Since log.hh is still included in sharded.hh, it will also solve the problem for the user of sharded.hh who assumed that seastar_logger is defined.

No, log.hh is for people who want to use their own loggers.  seastar_logger is internal.